src/mouse_pos.rs

```diff
@@ -76,16 +126,48 @@ impl Deref for MousePosWorld {
 }
 
 fn update_pos_ortho(
-    mut mouse_world: ResMut<MousePosWorld>,
-    mut cursor_moved: EventReader<CursorMoved>,
-    cameras: Query<(&GlobalTransform, &OrthographicProjection), With<Camera>>,
+    mut tracking: Query<(Entity, &mut MousePosWorld, &MousePos), Changed<MousePos>>,
+    cameras: Query<(&GlobalTransform, &OrthographicProjection)>,
 ) {
-    if let Some(event) = cursor_moved.iter().next_back() {
+    for (camera, mut world, screen) in tracking.iter_mut() {
         let (camera, proj) = cameras
-            .iter()
-            .next()
-            .expect("could not find an orthographic camera");
-        mouse_world.0 = camera
-            .mul_vec3(event.position.extend(0.0) + Vec3::new(proj.left, proj.bottom, proj.near));
+            .get(camera)
+            .expect("only orthographic cameras are supported");
+        world.0 =
+            camera.mul_vec3(screen.0.extend(0.0) + Vec3::new(proj.left, proj.bottom, proj.near));
     }
 }
+
+/// Marker component for the primary camera in the world.
+/// If only one camera exists, this is optional.
+#[derive(Debug, Clone, Copy, Component)]
+pub struct MainCamera;
+
+fn update_main_camera(
+    mut screen_res: ResMut<MousePos>,
+    mut world_res: ResMut<MousePosWorld>,
+    cameras: Query<(&MousePos, &MousePosWorld, Option<&MainCamera>)>,
+) {
+    use bevy::ecs::system::QuerySingleError;
+    let (screen, world, ..) = match cameras.get_single() {
+        Ok(x) => x,
+        Err(QuerySingleError::NoEntities(_)) => {
+            // this is okay, try again next frame.
+            return;
+        }
+        Err(QuerySingleError::MultipleEntities(_)) => {
+            // try to disambiguate
+            let mut mains = cameras.iter().filter(|(.., main)| main.is_some());
+
+            let main = mains.next().unwrap_or_else(||panic!("cannot identify main camera -- consider adding the MainCamera component to one of the cameras") );
+            if mains.next().is_some() {
```

Comment:
If we always want there to be exactly one or none of something in an ECS, a resource seems like a better model than iterating over a query every frame. This is already a backwards incompatible change (since the user has to choose `SingleCamera` or `MultiCamera` so it might be better to add the camera entity to a `MainCamera(pub Entity)` resource and then just throw an error here if the entity isn't a camera.

Reply:
I'm not too sure about requiring the user to initialize a MainCamera resource in order for the app to work -- I feel like `Res`ources in bevy tend to "just work". I wouldn't like for there to be cryptic panics.  

However, I have implemented something similar to this suggestion, where there is an _internal_ resource that tracks the current main camera. It gets updated only when `Camera` or `MainCamera` are changed, so the system doesn't have to figure out the main camera each and every frame.  

I may end up using this idea later, but it's out of scope of this PR.